lego/apps/surveys/authentication.py

```diff
@@ -0,0 +1,22 @@
+from rest_framework import authentication
+
+from lego.apps.surveys.models import Survey
+from lego.apps.users.models import User
+
+
+class SurveyAuthentication(authentication.BaseAuthentication):
+    def authenticate(self, request):
+
+        raw_token = request.GET.get('token')
+        if not raw_token:
+            return
+
+        try:
+            Survey.objects.get(token=raw_token)
+            return User.objects.get(
+                pk=1
+            ), raw_token  # TODO: Select a prod user for this or solve differently
```

Comment:
Gonna need some feedback on this. `authenticate` needs to return a tuple of (user, token) for authentication to be successful, but what user should be associated with people trying to access survey results by using a token? This `pk=1` referred to a test user without any permissions in my testing, and that worked fine. Is it an acceptable solution to make an anonymous user for this purpose and use his pk here? Or can we somehow work around this problem some other way? @eirsyl @hanswilw @odinuge